experimental/relativetime/src/relativetime.rs

```diff
@@ -0,0 +1,130 @@
+/// );
+/// assert_writeable_eq!(
+///         relative_time_formatter.format(-10i8),
+///         "10 seconds ago"
+/// );
+/// ```
+///
+/// # Example
+///
+/// ```
+/// use icu_relativetime::{RelativeTimeFormatter, RelativeTimeFormatterOptions};
+/// use icu_relativetime::options::Numeric;
+/// use icu_locid::locale;
+/// use writeable::assert_writeable_eq;
+///
+/// let relative_time_formatter = RelativeTimeFormatter::try_new_long_second_unstable(
+///     &icu_testdata::unstable(),
+///     &locale!("es").into(),
+///     RelativeTimeFormatterOptions { numeric: Numeric::Auto }
+/// )
+/// .expect("Data should load successfully.");
+///
+/// assert_writeable_eq!(
+///         relative_time_formatter.format(0u8),
+///         "ahora"
+/// );
+/// assert_writeable_eq!(
+///         relative_time_formatter.format(2u8),
+///         "dentro de 2 segundos"
+/// );
+/// assert_writeable_eq!(
+///         relative_time_formatter.format(-15i8),
+///         "hace 15 segundos"
+/// );
+/// ```
+pub struct RelativeTimeFormatter {
+    pub(crate) plural_rules: PluralRules,
+    pub(crate) rt: DataPayload<ErasedRelativeTimeFormatV1Marker>,
+    pub(crate) options: RelativeTimeFormatterOptions,
+    pub(crate) fixed_decimal_format: FixedDecimalFormatter,
+}
+
+macro_rules! constructor {
+    ($name: ident, $marker: ty) => {
+        #[doc = concat!("Create a new [`RelativeTimeFormatter`]")]
+        pub fn $name<D>(
+            data_provider: &D,
+            locale: &DataLocale,
+            options: RelativeTimeFormatterOptions,
+        ) -> Result<Self, RelativeTimeError>
+        where
+            D: DataProvider<CardinalV1Marker>
+                + DataProvider<$marker>
+                + DataProvider<DecimalSymbolsV1Marker>
+                + ?Sized,
+        {
+            let plural_rules = PluralRules::try_new_cardinal_unstable(data_provider, locale)?;
+            // Initialize FixedDecimalFormatter with default options
+            let fixed_decimal_format = FixedDecimalFormatter::try_new_unstable(
+                data_provider,
+                locale,
+                FixedDecimalFormatterOptions::default(),
+            )?;
+            let rt: DataPayload<$marker> = data_provider
+                .load(DataRequest {
+                    locale,
+                    metadata: Default::default(),
+                })?
+                .take_payload()?;
+            let rt = rt.cast();
+            Ok(RelativeTimeFormatter {
+                plural_rules,
+                options,
+                rt,
+                fixed_decimal_format,
+            })
+        }
+    };
+}
+
+impl RelativeTimeFormatter {
+    constructor!(
+        try_new_long_second_unstable,
+        LongSecondRelativeTimeFormatDataV1Marker
+    );
+
+    /// Format a `value` according to the locale and formatting options of
+    /// [`RelativeTimeFormatter`].
+    pub fn format<D: Into<FixedDecimal>>(&self, value: D) -> FormattedRelativeTime<'_> {
```

Comment:
Discussion: I tend to prefer being explicit about the type we take, rather than using `Into` bounds.

Reply:
I thought of having a `format_fixed_decimal` fn but that doesn't bring much value. Using `FixedDecimal` in `format` fn instead.